addons/l10n_tr_nilvera_einvoice/models/account_move.py

```diff
@@ -0,0 +1,240 @@
+        """Generic function to handle e-invoice and e-archive submissions to Nilvera."""
+        client = _get_nilvera_client(self.env.company)
+        response = client.request(
+            "POST",
+            endpoint,
+            files={'file': (xml_file.name, xml_file, 'application/xml')},
+            handle_response=False,
+        )
+
+        if response.status_code == 200:
+            self.l10n_tr_nilvera_send_status = 'sent'
+        elif response.status_code in [401, 403]:
+            raise UserError(_("Oops, seems like you're unauthorised to do this. Try another API key with more rights or contact Nilvera."))
+        elif 400 <= response.status_code < 500:
+            error_message, error_codes = self._l10n_tr_nilvera_einvoice_get_error_messages_from_response(response)
+
+            # If the sequence/series is not found on Nilvera, add it then retry.
+            if 3009 in error_codes and post_series_func:
+                post_series_func(client)
+                # post_series_func is not passed this time to avoid endless retries.
+                return self._l10n_tr_nilvera_submit_document(xml_file, endpoint)
+            raise UserError(error_message)
+        elif response.status_code == 500:
+            return UserError(_("Server error from Nilvera, please try again later."))
+
+        self.message_post(body=_("The invoice has been successfully sent to Nilvera."))
+
+    def _l10n_tr_nilvera_post_einvoice_series(self, client):
+        self._l10n_tr_nilvera_post_series("/einvoice/Series", client)
+
+    def _l10n_tr_nilvera_post_earchive_series(self, client):
+        self._l10n_tr_nilvera_post_series("/earchive/Series", client)
+
+    def _l10n_tr_nilvera_post_series(self, endpoint, client):
+        series = self.sequence_prefix.split('/')[0]
+        client.request(
+            "POST",
+            endpoint,
+            json={
+                'Name': series,
+                'IsActive': True,
+                'IsDefault': False,
+            },
+        )
+
+    def _l10n_tr_nilvera_get_submitted_document_status(self):
+        client = _get_nilvera_client(self.env.company)
+        for invoice in self:
+            response = client.request(
+                "GET",
+                f"/einvoice/sale/{invoice.l10n_tr_nilvera_uuid}/Status",
+            )
+
+            nilvera_status = response.get('InvoiceStatus', {}).get('Code')
+            if nilvera_status in dict(invoice._fields['l10n_tr_nilvera_send_status'].selection):
+                invoice.l10n_tr_nilvera_send_status = nilvera_status
+                if nilvera_status == 'error':
+                    msg = _("The invoice couldn't be sent to the recipient.<br/>")
+                    msg += "%s - %s<br/>" % (response['InvoiceStatus'].get('Description'), response['InvoiceStatus'].get('DetailDescription'))
+                    invoice.message_post(body=Markup(msg))
```

Comment:
Although the descriptions come from a trusted party, we're better of escaping them just in case. Furthermore, it's always better to escape translated strings in markups :+1: 

```suggestion
                    invoice.message_post(body=Markup(
                        "%s<br/>%s - %s<br/>" % (
                            _("The invoice couldn't be sent to the recipient.),
                            response['InvoiceStatus'].get('Description'),
                            response['InvoiceStatus'].get('DetailDescription'),
                        )
                    ))
```